### PR TITLE
Issue#5: add feature see Pet in Detail with Get-Endpoint

### PR DESCRIPTION
**Pull request by Anmavel**

### CODE REVIEW

File: backend/src/main/java/com/example/backend/service/PetService.java [hunk 1]
```diff
@@ -42,7 +45,7 @@ public List<Pet> deletePet(String id) {
             throw new PetNotFoundException("Pet with id"+ id+ "doesn't exist");
         }
         petRepo.deleteById(id);
-        return petRepo.findAll();
+        return getAllPets();
```
Comment (JohannesThorbergsson) [inline backend/src/main/java/com/example/backend/service/PetService.java:48]: This is non-standard. I suggest returning the deleted pet instead.

File: backend/src/test/java/com/example/backend/controller/PetControllerTest.java [hunk 2]
```diff
@@ -65,12 +65,31 @@ void when_addPet_then_OK() throws Exception {
...
 
+    @Test
+    @DirtiesContext
+    void when_getPetById_then_OK() throws Exception {
+        petRepo.save(pet1);
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/pets/1"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                        {"id":"1","name":"Whiskers","nameOfBreed":"albino", "photo":"albino.png","supplies": ["Water Bottle","Roomy Cage"]}          
+                        """));
```
Comment (JohannesThorbergsson) [inline backend/src/test/java/com/example/backend/controller/PetControllerTest.java:82]: This is difficult to read. I suggest to put each json-property in one line.

File: backend/src/main/java/com/example/backend/exception/GlobalExceptionHandler.java
Comment (JohannesThorbergsson) [inline backend/src/main/java/com/example/backend/exception/GlobalExceptionHandler.java:1]: Smart exception handling :-)

**Review by JohannesThorbergsson — APPROVED** [review]
lgtm :-)

File: backend/src/test/java/com/example/backend/controller/PetControllerTest.java (hunk 2 above)
Comment (Anmavel) [reply backend/src/test/java/com/example/backend/controller/PetControllerTest.java:82]: Thanks a lot :-)

**Pull request merged**